Several test related components are not available on Solaris; those
include flaky, pytest-mock and pytest-freezegun.

The pytest-django plugin is causing several tests to fail due
to changes in the working directory, so it is disabled.

Not intended for upstream.

--- virtualenv-20.29.2/pyproject.toml
+++ virtualenv-20.29.2/pyproject.toml
@@ -172,7 +172,7 @@ markers = [
   "slow",
 ]
 timeout = 600
-addopts = "--showlocals --no-success-flaky-report"
+addopts = "--showlocals -p no:django"
 env = [
   "PYTHONIOENCODING=utf-8",
 ]
--- virtualenv-20.29.2/tests/integration/test_zipapp.py
+++ virtualenv-20.29.2/tests/integration/test_zipapp.py
@@ -6,7 +6,6 @@ from contextlib import suppress
 from pathlib import Path
 
 import pytest
-from flaky import flaky
 
 from virtualenv.discovery.py_info import PythonInfo
 from virtualenv.info import fs_supports_symlink
@@ -105,7 +104,6 @@ def test_zipapp_in_symlink(capsys, call_
     assert not err
 
 
-@flaky(max_runs=2, min_passes=1)
 def test_zipapp_help(call_zipapp, capsys):
     call_zipapp("-h")
     _out, err = capsys.readouterr()
--- virtualenv-20.29.2/tests/conftest.py
+++ virtualenv-20.29.2/tests/conftest.py
@@ -366,3 +366,11 @@ else:
     @pytest.fixture
     def time_freeze(time_machine):
         return lambda s: time_machine.move_to(s, tick=False)
+
+@pytest.fixture()
+def freezer(*args, **kwargs):
+    pytest.skip("pytest-freezegun is not available on Solaris")
+
+@pytest.fixture()
+def mocker(*args, **kwargs):
+    pytest.skip("pytest-mock is not available on Solaris")
--- virtualenv-20.29.2/tests/unit/seed/embed/test_bootstrap_link_via_app_data.py
+++ virtualenv-20.29.2/tests/unit/seed/embed/test_bootstrap_link_via_app_data.py
@@ -215,6 +215,7 @@ def test_base_bootstrap_link_via_app_dat
         assert (result.creator.purelib / key).exists()
 
 
+@pytest.mark.skip(reason="Sometimes fails for unknown reasons.")
 @pytest.mark.usefixtures("temp_app_data")
 def test_app_data_parallel_ok(tmp_path):
     exceptions = _run_parallel_threads(tmp_path)
